src/main/java/io/cryostat/agent/model/MBeanInfo.java

```diff
@@ -0,0 +1,211 @@
+/*
+ * Copyright The Cryostat Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.cryostat.agent.model;
+
+import java.io.ByteArrayOutputStream;
+import java.io.DataOutputStream;
+import java.io.IOException;
+import java.lang.management.ManagementFactory;
+import java.lang.management.MemoryMXBean;
+import java.lang.management.OperatingSystemMXBean;
+import java.lang.management.RuntimeMXBean;
+import java.lang.management.ThreadMXBean;
+import java.nio.charset.StandardCharsets;
+import java.util.Arrays;
+import java.util.Base64;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.Callable;
+
+import io.cryostat.core.net.MBeanMetrics;
+import io.cryostat.core.net.MemoryMetrics;
+import io.cryostat.core.net.OperatingSystemMetrics;
+import io.cryostat.core.net.RuntimeMetrics;
+import io.cryostat.core.net.ThreadMetrics;
+
+import com.sun.management.UnixOperatingSystemMXBean;
+import org.apache.commons.codec.digest.DigestUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class MBeanInfo {
+
+    private final Logger log = LoggerFactory.getLogger(getClass());
+    private MBeanMetrics mBeanMetrics;
+    private Map<String, Object> simplifiedMetrics;
```

Comment:
The way MBeanMetrics is put together it would be a bit cumbersome to reconstruct the map that gets sent to the CEL script evaluator on demand, it would essentially be reimplementing the other getMetrics methods in the class. I can change it if you'd prefer to save the memory, but we'd end up essentially duplicating a lot of code either way.